src/components/cc-oauth-consumer/cc-oauth-consumer.smart.js

```diff
@@ -0,0 +1,82 @@
+// @ts-expect-error FIXME: remove when clever-client exports types
+import { get } from '@clevercloud/client/esm/api/v2/oauth-consumer.js';
+import { sendToApi } from '../../lib/send-to-api.js';
+import { defineSmartComponent } from '../../lib/smart/define-smart-component.js';
+import '../cc-smart-container/cc-smart-container.js';
+import './cc-oauth-consumer.js';
+import { camelCase } from '../../lib/change-case.js';
+
+/**
+ * @typedef {import('./cc-oauth-consumer.js').CcOauthConsumer} CcOauthConsumer
+ * @typedef {import('./cc-oauth-consumer.types.js').OauthConsumerStateLoaded} OauthConsumerStateLoaded
+ * @typedef {import('../../lib/send-to-api.types.js').ApiConfig} ApiConfig
+ * @typedef {import('../../lib/smart/smart-component.types.js').OnContextUpdateArgs<CcOauthConsumer>} OnContextUpdateArgs
+ */
+
+defineSmartComponent({
+  selector: 'cc-oauth-consumer',
+  params: {
+    apiConfig: { type: Object },
+    ownerId: { type: String },
+    key: { type: String },
+  },
+
+  /**
+   * @param {OnContextUpdateArgs} args
+   */
+  onContextUpdate({ context, updateComponent }) {
+    const { apiConfig, ownerId, key } = context;
+    const api = new Api(apiConfig, ownerId, key);
+
+    updateComponent('state', { type: 'loading' });
+
+    api
+      .getOauthConsumer()
+
+      .then(
+        /** @param {OauthConsumerStateLoaded} data */
```

Comment:
nitpick: can we move this type info to the `@return` of `getOauthConsumer` on `class Api`?